go/vt/vtgate/planbuilder/route.go

```diff
@@ -337,7 +346,17 @@ func (rb *route) PushSelect(expr *sqlparser.NonStarExpr, _ *route) (colsym *cols
 // PushStar pushes the '*' expression into the route.
 func (rb *route) PushStar(expr *sqlparser.StarExpr) *colsym {
 	colsym := newColsym(rb, rb.Symtab())
-	colsym.Alias = sqlparser.NewColIdent(sqlparser.String(expr))
+	// This is not perfect, but it should be good enough.
+	// We'll match unqualified column names against Alias
+	// and qualified column names against QualifiedName.
+	// If someone uses 'select *' and then uses table.col
+	// in the HAVING clause, then things won't match. But
+	// such cases are easy to correct in the application.
+	if expr.TableName == "" {
+		colsym.Alias = sqlparser.NewColIdent(sqlparser.String(expr))
+	} else {
+		colsym.QualifiedName = sqlparser.NewColIdent(sqlparser.String(expr))
```

Comment:
If you had `select a.*`, then Alias will be empty and QualifiedName will be `a.*`. In this situation references to `col` or `b.col` will fail to match. But `a.col` will match.